### PR TITLE
FIX no Armor Damage - #1962

### PR DESCRIPTION
**Pull request by laurobeleche**
### Description
To fix the bug of no Armor damage

### Reason to modify
I tested within my server initially created a plugin that corrected the bug, then I realized that the function that creates the damage the armor was not being called, then traced the causes that damage the armor and put to call the function.

Fix no Armor damage

### CODE REVIEW

File: src/pocketmine/event/entity/EntityDamageEvent.php [hunk 1]
```diff
@@ -159,6 +159,7 @@ public function __construct(Entity $entity, $cause, $damage){
 							}
 						}
 					}
+					$this->useArmors();
```
Comment (dktapps) [inline src/pocketmine/event/entity/EntityDamageEvent.php]: This is one reason I tell people not to use the web editor. That method does not exist in EntityDamageEvent nor any of its parents.

Comment (laurobeleche) [inline src/pocketmine/event/entity/EntityDamageEvent.php]: But then what editor I use, I have it on my PC and edit with Notepad ++, this change fixed the bug.

**Comment by laurobeleche** [comment]
`/**
	 * @return bool
	 */
	public function useArmors(){
		if($this->entity instanceof Player){
			if($this->entity->isSurvival() and $this->entity->isAlive()){
				foreach($this->usedArmors as $index => $cost){
					$i = $this->entity->getInventory()->getArmorItem($index);
					if($i->isArmor()){
						/** @var Armor $i */
						$i->useOn($i, $cost);
						$this->entity->getInventory()->setArmorItem($index, $i);
					}
				}
			}
			return true;
		}
		return false;
	}`
This function is rather the event on line 301 you can check the method, and it works, just be called as I did in my correction.

**Comment by dktapps** [comment]
>_> blame me for using a mobile device to review code. My bad, sorry.

File: src/pocketmine/event/entity/EntityDamageEvent.php (hunk 1 above)
Comment (dterweij) [inline src/pocketmine/event/entity/EntityDamageEvent.php]: PHPStorm IDE.

**Comment by dog194** [comment]
no no no... useArmors() will reduce the damage which is list in $this->usedArmors.
and you should not use this function in event!!!!!!!!!
this function should only trigger when the player really get hurt. 

e.g. if a player walk into a Fire block. This event will be called more than 10 times in 1 second. With player have damage cooldown. so just one of this event will really hurt the player.!!!
reference: https://github.com/iTXTech/Genisys/search?utf8=%E2%9C%93&q=useArmors%28%29

BTW i am really busy recent days. QAQ so for this

**Comment by dog194** [comment]
fewer months ago, when this event change. Armors Durability System work fine with walking into fire lava and Cactus. So maybe you should have a check useArmors() function. 

I'm sorry that I can not help due to a  ton of my daily work.

**Comment by laurobeleche** [comment]
but then why the function has not been called in the case of PVP damage?

**Comment by dog194** [comment]
@laurobeleche     
src/pocketmine/Player.php   line 3013

**Comment by laurobeleche** [comment]
But the function has not been executed, the bug's there a long time and NOBODY is doing nothing to fix it. I mobilized, I opened the door, showed the way, I have little experience with pocketmine / Genisys, but I will delve to remedy the various bugs that harm us every day.

**Comment by laurobeleche** [comment]
In my tests, I identified the
if ($ target-> attack (EV $> getFinalDamage (), $ ev) === true)
There was returning TRUE in any way, then I added a true return to function so that the condition was true.
Thanks @dog194 @dktapps

**Comment by SuperDude36** [comment]
I downloaded Genisys on Jenkins and armor still doesn't break.

**Comment by laurobeleche** [comment]
https://gitlab.com/laurobeleche/Genisys/builds/5078359/artifacts/download

Download this,

**Comment by SuperDude36** [comment]
Eh, 404 error.

**Comment by laurobeleche** [comment]
https://gitlab.com/laurobeleche/Genisys/pipelines

Commit players.php, download artfacts

**Comment by SuperDude36** [comment]
Still 404 error...

**Comment by laurobeleche** [comment]
Sorry

http://www.mediafire.com/file/zal9m7xu19m1zdr/Genisys_bef53a6_2016-10-13_02-18-58.phar

**Comment by dog194** [comment]
errrrr. For some reason i cannot help you to have a check now. 
So maybe merge until getting more than 3 test result from other user~.
Note:   PVP  walk into fire block and so on
need to check the armor only lost 1 durability when get hurt.

**Comment by SuperDude36** [comment]
Thanks for the media fire link!!! Works!

**Comment by ishitatsuyuki** [comment]
You created redundant conflicts. Learn to use git.

**Comment by laurobeleche** [comment]
Sorry

**Comment by ishitatsuyuki** [comment]
**Use Git or close.**

**Comment by laurobeleche** [comment]
Sorry, I installed git and now I started working with him.

**Pull request closed without merging**